source/connection-monitoring-and-pooling/tests/pool-checkout-maxConnecting-is-enforced.yml

```diff
@@ -0,0 +1,75 @@
+version: 1
+style: integration
+description: maxConnecting is enforced
+runOn:
+  -
+    # required for blockConnection in fail point
+    minServerVersion: "4.4.0"
+failPoint:
+  configureFailPoint: failCommand
+  mode: { times: 3 }
+  data:
+    failCommands: ["isMaster"]
+    closeConnection: false
+    blockConnection: true
+    blockTimeMS: 750
+poolOptions:
+  maxPoolSize: 10
+  waitQueueTimeoutMS: 5000
+operations:
+  # start creating a Connection. This will take a while
+  # due to the fail point.
+  - name: start
+    target: thread1
+  - name: checkOut
+    thread: thread1
+  # Start 2 new threads that wait for a little then try
+  # to check out connections. Only one thread should
+  # start creating a Connection and the other one should be
+  # waiting for pendingConnectionCount to be less than maxConnecting,
+  # only starting once thread1 finishes creating its Connection.
+  - name: start
+    target: thread2
+  - name: wait
+    thread: thread2
+    ms: 100
+  - name: checkOut
+    thread: thread2
+  - name: start
+    target: thread3
+  - name: wait
+    thread: thread3
+    ms: 100
+  - name: checkOut
+    thread: thread3
+  # wait until all Connections have been created.
+  - name: waitForEvent
+    event: ConnectionReady
+    count: 3
+events:
```

Comment:
Have you implemented these tests? They are consistently failing for me. For example in this case the actual events I get before filtering out ignored events are:
```
PoolCreatedEvent(('localhost', 27017), {'maxPoolSize': 10, 'waitQueueTimeoutMS': 5000.0})
ConnectionCheckOutStartedEvent(('localhost', 27017))
ConnectionCreatedEvent(('localhost', 27017), 1)
ConnectionReadyEvent(('localhost', 27017), 1)
ConnectionCheckedOutEvent(('localhost', 27017), 1)
ConnectionCheckOutStartedEvent(('localhost', 27017))
ConnectionCreatedEvent(('localhost', 27017), 2)
ConnectionReadyEvent(('localhost', 27017), 2)
ConnectionCheckedOutEvent(('localhost', 27017), 2)
ConnectionCheckOutStartedEvent(('localhost', 27017))
ConnectionCreatedEvent(('localhost', 27017), 3)
ConnectionReadyEvent(('localhost', 27017), 3)
ConnectionCheckedOutEvent(('localhost', 27017), 3)
```

Interestingly, when I increase the failCommand `times` from 3 to 20, the test succeeds. This indicates that other connections are consuming the failpoint. In python's test suite there are many clients running at a time. Ideally we could use appName to restrict this fail point to only a single client but that is blocked by this bug: https://jira.mongodb.org/browse/SERVER-49220

Can you increase  `times`?

Reply:
> We don't use clients to run the CMAP tests, so we don't have any interference coming from background monitors

In python we have to use a MongoClient because that's the entity which handles the minPoolSize/maxIdleTime requirements. Besides that we also have various other shared MongoClients open, for example our test suite creates one meta client that stays open for all tests. This meta client can inadvertently trigger the failpoint. 

> Should I set it to alwaysOn to ensure there's absolutely no number of concurrent clients that could interfere? Or is it safer to specify a high-but-finite value?

high-but-finite value is safe imo.

Reply:
Increased `times` to 10.

I eagerly await the resolution of SERVER-49220. This seems like a dirty solution lol.